pkg/cmd/server/origin/controller/network_linux.go

```diff
@@ -18,13 +23,22 @@ func (c *SDNControllerConfig) RunController(ctx ControllerContext) (bool, error)
 		return false, nil
 	}
 
+	networkClient := ctx.ClientBuilder.OpenshiftInternalNetworkClientOrDie(bootstrappolicy.InfraSDNControllerServiceAccountName)
+	sdnInformers := sdncommon.SDNInformers{
+		KubeInformers:    ctx.InternalKubeInformers,
+		NetworkInformers: networkinformers.NewSharedInformerFactory(networkClient, networkapi.DefaultInformerResyncPeriod),
```

Comment:
All the other OpenShift API groups already have an Informers field in ControllerContext. We should make NetworkInformers work the same way. (Add a field to ControllerContext and initialize it (and eventually call Start() on it) from pkg/cmd/server/start/.)

Reply:
I originally took this route. Adding a new field in existing informers struct in start/informers.go. All the informers are initialized in NewInformers() and passed to newCreateControllerContext() and ControllerContext runs all controllers(including SDN). But the problem is we don't want to run our network informers if the user is not using openshift-sdn, so I started sprinkling IsOpenShiftNetworkPlugin() in few places. Then I realized why NewInformers/ControllerContext() need to know about openshift network plugin? SDN RunController() is the starting point for openshift sdn and putting network informers there seems the right place.

Reply:
Hm... it's weird to treat pkg/networking/apis differently from all the other API groups, but then again, pkg/networking/apis *is* different from all the other API groups; it's only used by our network plugins, not by the OpenShift core or by third-party network plugins. So, OK, I guess this makes sense.